### PR TITLE
Improve performance of internal traces to jaeger proto translation

### PR DESCRIPTION
**Pull request by dmitryax**
Benchmark results:

Before:
```
BenchmarkInternalTracesToJaegerProto-12       611580          1696 ns/op        2952 B/op         17 allocs/op
```

After:
```
BenchmarkInternalTracesToJaegerProto-12       837252          1299 ns/op        2024 B/op         13 allocs/op
```

### CODE REVIEW

File: translator/trace/jaeger/traces_to_jaegerproto.go [hunk 1]
```diff
@@ -115,42 +115,64 @@ func resourceToJaegerProtoProcess(resource pdata.Resource) *model.Process {
...
+	}
+
+	tags := make([]model.KeyValue, 0, attrsCount)
+	process.Tags = appendTagsFromResourceAttributes(tags, attrs)
 	return &process
 
 }
 
-func attributesToJaegerProtoTags(attrs pdata.AttributeMap) []model.KeyValue {
+func appendTagsFromResourceAttributes(dest []model.KeyValue, attrs pdata.AttributeMap) []model.KeyValue {
```
Comment (bogdandrutu) [inline translator/trace/jaeger/traces_to_jaegerproto.go:133]: Simpler code:
```
process.Tags = appendTagsFromResourceAttributes(attrsCount, attrs)
return &process

func appendTagsFromResourceAttributes(attrs pdata.AttributeMap) []model.KeyValue {
	if attrs.Cap() == 0 {
		return nil
	}
        dest = make([]model.KeyValue, 0, attrs.Cap())
}
```

File: translator/trace/jaeger/traces_to_jaegerproto.go [hunk 2]
```diff
@@ -115,42 +115,64 @@ func resourceToJaegerProtoProcess(resource pdata.Resource) *model.Process {
 	}
 
 	process := model.Process{}
+	attrsCount := attrs.Cap()
 	if serviceName, ok := attrs.Get(conventions.AttributeServiceName); ok {
 		process.ServiceName = serviceName.StringVal()
-		attrs.Delete(conventions.AttributeServiceName)
+		attrsCount--
```
Comment (bogdandrutu) [inline translator/trace/jaeger/traces_to_jaegerproto.go:121]: Please fix the bug in a separate PR since this is very confusing now.

Comment (bogdandrutu) [inline translator/trace/jaeger/traces_to_jaegerproto.go:121]: Also not sure I understand why do you call `attrsCount--`. If the attrs will still have `service.name` in the `appendTagsFromResourceAttributes` you will append the `service.name` as well so not sure then why allocating only a slice of (attrs.Cap() - 1).

Comment (dmitryax) [reply translator/trace/jaeger/traces_to_jaegerproto.go:121]: `appendTagsFromResourceAttributes` does have the logic to skip the service name attribute https://github.com/open-telemetry/opentelemetry-collector/pull/906/files#diff-645cbe27bc5be6692575be50e2b28640R139

Comment (bogdandrutu) [reply translator/trace/jaeger/traces_to_jaegerproto.go:121]: Saw that after.

Comment (dmitryax) [reply translator/trace/jaeger/traces_to_jaegerproto.go:121]: Submitter another PR https://github.com/open-telemetry/opentelemetry-collector/pull/907, please take a look, I'll rebase once it's merged

File: translator/trace/jaeger/traces_to_jaegerproto.go (hunk 1 above)
Comment (dmitryax) [reply translator/trace/jaeger/traces_to_jaegerproto.go:133]: It's not always the case that we can use `attrs.Cap()` as capacity for tags slice, that's why I moved it outside

Comment (bogdandrutu) [reply translator/trace/jaeger/traces_to_jaegerproto.go]: Hmm, confused, why, because we allocate 1 extra element?

**Comment by dmitryax** [comment]
@bogdandrutu PTAL

File: translator/trace/jaeger/traces_to_jaegerproto.go (hunk 1 above)
Comment (dmitryax) [reply translator/trace/jaeger/traces_to_jaegerproto.go]: The similar function `appendTagsFromAttributes` needs to allocate more than `attrs.Cap()`, I kept the same approach here as well

**Pull request merged**